ssh/client.go

```diff
@@ -0,0 +1,153 @@
+package ssh
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+type Client struct {
+	Config   *ssh.ClientConfig
+	Hostname string
+	Port     int
+}
+
+func NewClient(user string, host string, port int, auth *Auth) (*Client, error) {
+	config, err := NewConfig(user, auth)
+	if err != nil {
+		return nil, err
+	}
+
+	client := &Client{
+		Config:   config,
+		Hostname: host,
+		Port:     port,
+	}
+
+	return client, nil
+}
+
+func NewConfig(user string, auth *Auth) (*ssh.ClientConfig, error) {
+	var authMethods []ssh.AuthMethod
+
+	for _, k := range auth.Keys {
+		key, err := ioutil.ReadFile(k)
+		if err != nil {
+			return nil, err
+		}
+
+		privateKey, err := ssh.ParsePrivateKey(key)
+		if err != nil {
+			return nil, err
+		}
+
+		authMethods = append(authMethods, ssh.PublicKeys(privateKey))
+	}
+
+	for _, p := range auth.Passwords {
+		authMethods = append(authMethods, ssh.Password(p))
+	}
+
+	return &ssh.ClientConfig{
+		User: user,
+		Auth: authMethods,
+	}, nil
+}
+
+func (client *Client) Run(command string) (*Output, error) {
+	conn, err := ssh.Dial("tcp", fmt.Sprintf("%s:%d", client.Hostname, client.Port), client.Config)
+	if err != nil {
+		return nil, err
+	}
+
+	session, err := conn.NewSession()
+	if err != nil {
+		return nil, err
+	}
+
+	defer session.Close()
+
+	var stdout, stderr bytes.Buffer
+
+	session.Stdout = &stdout
+	session.Stderr = &stderr
+
+	if err := session.Run(command); err != nil {
+		return nil, err
+	}
+
+	return &Output{
+		Stdout: &stdout,
+		Stderr: &stderr,
+	}, nil
+}
+
+func (client *Client) Shell() error {
+	conn, err := ssh.Dial("tcp", fmt.Sprintf("%s:%d", client.Hostname, client.Port), client.Config)
+	if err != nil {
+		return err
+	}
+
+	session, err := conn.NewSession()
+	if err != nil {
+		return err
+	}
+
+	defer session.Close()
+
+	session.Stdout = os.Stdout
```

Comment:
Ah my bad I didnt see NewSession above.  Actually if we always set stdin stdout on session maybe move these to NewSession as parameters?